assembl/static2/js/app/components/cookies/acceptCookiesModal.jsx

```diff
@@ -0,0 +1,190 @@
+// @flow
+import React from 'react';
+import { compose, graphql } from 'react-apollo';
+import { connect } from 'react-redux';
+import { Modal, FormGroup, Checkbox, Button } from 'react-bootstrap';
+import { Translate } from 'react-redux-i18n';
+
+import { legalContentSlugs, ESSENTIAL_SIGNUP_COOKIES as LEGAL_CONTENTS_TO_ACCEPT } from '../../constants';
+import manageErrorAndLoading from './../../components/common/manageErrorAndLoading';
+import { getRouteLastString, getDiscussionSlug } from '../../utils/globalFunctions';
+import { get, getContextual } from '../../utils/routeMap';
+
+import LegalContentsLinksList from './legalContentsLinksList';
+
+// GraphQL imports
+import TabsConditionQuery from './../../graphql/TabsConditionQuery.graphql';
+import updateAcceptedCookies from './../../graphql/mutations/updateAcceptedCookies.graphql';
+import acceptedCookiesQuery from './../../graphql/acceptedCookiesQuery.graphql';
+
+type LegalContentsArray = Array<string>;
+
+type Props = {
+  pathname: string,
+  userId: string,
+  hasTermsAndConditions: boolean,
+  hasPrivacyPolicy: boolean,
+  hasUserGuidelines: boolean,
+  acceptedLegalContentsList: LegalContentsArray,
+  updateAcceptedCookies: Function
+};
+
+type State = {
+  showModal: boolean,
+  modalIsChecked: boolean
+};
+
+export class DumbAcceptCookiesModal extends React.PureComponent<Props, State> {
+  state = {
+    modalIsChecked: false,
+    showModal: false
+  };
+
+  componentDidMount() {
+    this.showModal();
+  }
+
+  showModal = () => {
+    const { userId, pathname, acceptedLegalContentsList } = this.props;
+    const lastRouteString = getRouteLastString(pathname);
+    const isOnLegalContentPage = legalContentSlugs.includes(lastRouteString);
+    let userHasAcceptedAllLegalContents;
+    // This array gathers all the legal contents to accept by their 'ACCEPT_...' formatted name
+    const legalContentsToAcceptByCookieName = this.getLegalContentsToAccept();
+    if (userId) {
+      userHasAcceptedAllLegalContents = legalContentsToAcceptByCookieName.every(legalContent =>
+        acceptedLegalContentsList.includes(legalContent)
+      );
+    }
+    // The modal is only showed to a user who is connected but hasn't yet accepted legal contents and isn't currently reading them
+    if (!isOnLegalContentPage && !userHasAcceptedAllLegalContents && userId) {
+      this.setState({ showModal: true });
+    }
+  };
+
+  handleModalCheckbox = () => {
+    this.setState(prevState => ({
+      modalIsChecked: !prevState.modalIsChecked
```

Comment:
it's not the modal that is checked, right? Maybe `acceptButtonEnabled`? (or `acceptButtonDisabled` and change the code)

Reply:
modalCheckboxIsChecked ? 
The fact that the button is enabled or not is a consequence of this piece of state, not what it fundamentally represent